src/Sylius/Component/Promotion/Factory/ActionFactory.php

```diff
@@ -63,4 +63,16 @@ public function createPercentageDiscount($percentage)
 
         return $action;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createPercentageShippingDiscount($percentage)
+    {
+        $action = $this->createNew();
+        $action->setType('shipping_discount');
```

Comment:
It should definitely be a constant, however I don't know when to put it :smile: Obvious choice is ``ShippingDiscountAction``, but it is in *Core* and for now ``ActionFactory`` is in *Promotion* component - which [should be changed](https://github.com/Sylius/Sylius/issues/4229).
So maybe the best solution would be to solve mentioned issue and then do it as it should be done ;)